src/utils/appData.spec.ts

```diff
@@ -1,89 +1,48 @@
-}
+import { getSerializedCID, loadIpfsFromCid } from './appData'
+import { DEFAULT_IPFS_READ_URI } from '../constants'
 
 const INVALID_CID_LENGTH = 'Incorrect length'
 
-describe('validateAppDataDocument', () => {
-  const v010Doc = {
-    ...BASE_DOCUMENT,
-    metatadata: {
-      referrer: { address: '0xb6BAd41ae76A11D10f7b0E664C5007b908bC77C9', version: '0.1.0' },
-    },
-  }
-  const v040Doc = {
-    ...v010Doc,
-    version: '0.4.0',
-    metadata: { ...v010Doc.metadata, quote: { slippageBips: '1', version: '0.2.0' } },
-  }
-
-  test('Version matches schema', async () => {
+describe('getSerializedCID', () => {
+  test('Serializes hash into CID', async () => {
     // given
-    // when
-    const v010Validation = await validateAppDataDocument(v010Doc, v010Doc.version)
-    const v040Validation = await validateAppDataDocument(v040Doc, v040Doc.version)
-    // then
-    expect(v010Validation).toEqual(VALID_RESULT)
-    expect(v040Validation).toEqual(VALID_RESULT)
-  })
+    const hash = '0xa6c81f4ca727252a05b108f1742a07430f28d474d2a3492d8f325746824d22e5'
+    const expected = 'QmZZhNnqMF1gRywNKnTPuZksX7rVjQgTT3TJAZ7R6VE3b2'
 
-  test("Version doesn't match schema", async () => {
-    // given
     // when
-    const v030Validation = await validateAppDataDocument(v040Doc, '0.3.0')
-    // then
-    expect(v030Validation.result).toBeFalsy()
-    expect(v030Validation.errors).toEqual("data/metadata/quote must have required property 'sellAmount'")
-  })
+    const cidV0 = await getSerializedCID(hash)
 
-  test("Version doesn't exist", async () => {
-    // given
-    // when
-    const validation = validateAppDataDocument(v010Doc, '0.0.0')
     // then
-    await expect(validation).rejects.toThrow('AppData version 0.0.0 does not exist')
+    expect(cidV0).toEqual(expected)
   })
-})
 
-describe('getSerializedCID', () => {
-  test('Valid serialized appData CID', async () => {
-    const hash = '0xa6c81f4ca727252a05b108f1742a07430f28d474d2a3492d8f325746824d22e5'
-    const serializedCidV0 = 'QmZZhNnqMF1gRywNKnTPuZksX7rVjQgTT3TJAZ7R6VE3b2'
-    const cidV0 = await getSerializedCID(hash)
+  test('Throws on invalid appData hash', async () => {
+    // given
+    const invalidHash = '0xa6c81f4ca727252a05b108f1742'
 
-    expect(cidV0).toEqual(serializedCidV0)
-  })
+    // when
+    const promise = getSerializedCID(invalidHash)
 
-  test('Invalid: serialized appData CID format ', async () => {
-    const invalidHash = '0xa6c81f4ca727252a05b108f1742'
-    try {
-      await getSerializedCID(invalidHash)
-    } catch (e: unknown) {
-      const error = e as Error
-      expect(error.message).toEqual(INVALID_CID_LENGTH)
-    }
+    // then
+    await expect(promise).rejects.toThrow(INVALID_CID_LENGTH)
   })
 })
 
 describe('loadIpfsFromCid', () => {
   test('Valid IPFS appData from CID', async () => {
-    fetchMock.mockResponseOnce(
+    // given
+    const validSerializedCidV0 = 'QmZZhNnqMF1gRywNKnTPuZksX7rVjQgTT3TJAZ7R6VE3b2'
+    const expected =
```

Comment:
why is this a string/JSON vs an object that doesn't need to be parsed below? i get that you're shaping the expected response as you would get it but we're not testing the parse function. Or is this a way of testing responses? (genuine question)

Reply:
`fetchMock.mockResponseOnce(expected)` expects a string or a function.
I feel it's cleaner to pass a string here.

The goal here is to test the `loadIpfsFromCid` returns the parsed version of the response given by the endpoint rather than making the actual network call.
The JSON.parse is just a convenience. I could do the other way around: store the object and JSON.stringify in the mock response and compare them both 🤷